### DISCUSSION TITLE
argocd kustomize git-secrets

### DISCUSSION THREAD

**Discussion by ggrames** [argoproj/argo-cd#15109] [Q&A] [answered]
Hi,
i am using argocd with kustomize repo which is stored in gitlab.
It is working fine for deployments, configmaps and so on.
My only blocking point is how to use / promote mit secrets, which are stored in gitlab (also in the kustomized repo).
The secrets are encrypted with [git-secret](https://sobolevn.me/git-secret/).
At the moment i have no idea how to decrypt them and kustomize it via argocd.
I had a look already to sealed secrets, but in this case it seems to be a seperate manual work to generate the secrets and maintain it offside kustomize.

Has somebody some further hints for me to archive this?

I am using:
argocd
    "Version": "v2.4.28+unknown"
which comes from openshift (4.9) GitOps Operator 

Thank you

**Comment by gaeljw** [comment, marked as answer]
You can probably write a custom plugin in ArgoCD but IMHO that's not the right direction: I would look to change the approach and use one which doesn't require anything at deployment step (I.e. ArgoCD) but rather use a controller on the cluster to transform/inject secrets like Sealed Secrets or External Secrets Operator.

**Reply by ggrames** [reply]
Ok, thank you for the info.
I think i will try to write a ArgoCD plugin for the git-secret encrypted secrets in my repo,
Do you know a detailed HowTo for writing plugins for Argo?
Thank you
